handlers/controller.js

```diff
@@ -7,28 +11,69 @@ const { models, connectDb } = require ('../dal/database');
  * @returns {Promise<void>} n/a
  */
 let ping = async (req, res) => {
-    res.status(200).send('Ping!');
+    res.status(StatusCode.EASTER_EGG).send('Ping!');
 };
 
 /**
- * Gets product from DAL.
+ * Gets product from DAL - if not found it is retrieved from Barcodelookup and then stored in MongoDB.
  * @param req request.
  * @param res response.
  * @returns {Promise<void>} n/a
  */
 let getProduct = async (req, res) => {
-    let barcodeId = req.params.id;
+    let barcode = req.params.id;
 
     connectDb().then(async () => {
-        models.Product.findOne({ barcodeId })
-            .then((doc) => {
-                if(!doc)
-                    return res.status(404).send();
-                else
-                    res.status(200).send({ doc });
-            }).catch((err) => res.status(400).send(err));
+        try {
+            // search for product in MongoDB
+            let doc = await models.Product.findOne({barcode});
+            if(doc) {
+                console.log(`found ${doc.barcode} in mongodb`);
+                return res.status(StatusCode.OK).send({doc});
+            }
+
+            // else query barcodelookup for product
+            let bclRes = await bc.lookup({key: process.env.API_KEY, barcode: barcode});
+            if(bclRes.statusCode !== StatusCode.OK) {
+                console.error(`error looking up ${barcode} in barcodelookup`);
+                return res.status(bclRes.statusCode).send({ data: bclRes.data });
+            }
+
+            let manufacturer;
+
+            if (bclRes.data.manufacturer)
+                manufacturer = bclRes.data.manufacturer;
+            else if (bclRes.data.brand)
+                manufacturer = bclRes.data.brand;
+            else
+                console.error(`cannot find a manufacturer for ${barcode}`);
```

Comment:
Cool, and perhaps we can later ask users to manually fill in only the missing bits. Not sure how difficult this would be.

Reply:
Easy work. We can address that in another pull request.